### ISSUE TITLE
About AttrField.extract_value

### ISSUE THREAD

**Issue by panhaoyu** [howie6879/ruia#14]
```python3
def extract_value(self, html, is_source=False):
       """
       Use css_select or re_select to extract a field value
       :return:
       """
       if self.css_select:
           value = html.cssselect(self.css_select)
           value = value[0].get(self.attr, value) if len(value) == 1 else value
       elif self.xpath_select:
           value = html.xpath(self.xpath_select)
       else:
           raise ValueError('%s field: css_select or xpath_select is expected' % self.__class__.__name__)
       if is_source:
           return value
       if self.default is not None:
           value = value if value else self.default
       return value
```

There's such a line:

```python3
value = value[0].get(self.attr, value) if len(value) == 1 else value
```

I have such an idea, to extract data from each item, and then check the count:

```python3
if self.css_select:
    value = html.cssselect(self.css_select)
    value = [item.get(self.attr, item) for item in value]
    if len(value) == 1:
    value = value[0]
```

Besides, I don't think it's necessary to check count.
It may cause confusion when some pages have several items while some pages have only one item.
For example, there is a field named 'tag'.
I think always return a list is better.

**Comment by howie6879** [comment]
Hi @panhaoyu : 
Thanks for your issue, about this question, i think it's necessary to check count.
If the length of the extracted value equal to 1, the value can be returned without writing a clean function

```python
from ruia import AttrField, TextField, Item


class HackerNewsItem(Item):
    target_item = TextField(css_select='tr.athing')
    title = TextField(css_select='a.storylink')

    async def clean_title(self, value):
       # Avoid use this method when the length is equal to 1
       return value[0]
```
I think your plan will work well , Can you submit a PR about this issue?

**Comment by panhaoyu** [comment]
hey, I'm sorry, may I confirm your view...
emm, I raised two issues here

1. add a `many=False` argument to fields
2. always return string

should I fix both the two?

It may cause a backward incompatibility.

**Comment by howie6879** [comment]
This issue fixed #15
